### ISSUE TITLE
Painless: improve error message when parameter must be a constant but isn't

### ISSUE THREAD

**Issue by nik9000** [elastic/elasticsearch#68324]
Now the error message is:
```
"all arguments must be constant but the [2] argument isn't
```

That isn't really even English. I mean, it tells you where to look and we don't have any better ideas given the infrastructure we have, but we do need to make it better.

**Comment by stu-elastic** [comment]
Yes this message should be better.  @nik9000 can you provide a brief script illustrating the problem?

**Comment by nik9000** [comment]
In a runtime field if you do something like:
```
String a = 'a';
String b = 'b';
grok(a + b);
```

Which, I think, will reply with something like: "all arguments must be constant but the [1] argument isn't"

**Comment by aileenpalafox** [comment]
Hi, can I check this out? First time contributing

**Comment by andrevpessanha** [comment]
Hi @nik9000, i opened a PR to solve this issue and since this is my first contribution, i'd be really glad if you could give me some feedback.

**Comment by andrevpessanha** [comment]
Hi @nik9000, could you tell me what do you think of these new error messages? I had mapped four occurrences of this type of error, but please let me know if i forgot to update any other messages.

**Comment by Kxrr** [comment]
Could I jump on this issue? 

I'm going to optimize the error message to:

```
All arguments of the [<methodName>] method must be constants, 
but the argument [<argument>] is not
```

**Comment by stu-elastic** [comment]
@Kxrr, That'd be a good message

**Comment by Kxrr** [comment]
Hi @stu-elastic, could you please review my changes in #82638

**Comment by jdconrad** [comment]
Looks like this was fixed by #68517
